executor/common_linux.h

```diff
@@ -4564,3 +4656,236 @@ static volatile long syz_fuse_handle_req(volatile long a0, // /dev/fuse fd.
+	netlink_init(nlmsg, nl80211_family, 0, &genlhdr, sizeof(genlhdr));
+	netlink_attr(nlmsg, NL80211_ATTR_IFINDEX, &ifindex, sizeof(uint32));
+	netlink_attr(nlmsg, NL80211_ATTR_SSID, props->ssid, props->ssid_len);
+	netlink_attr(nlmsg, NL80211_ATTR_WIPHY_FREQ, &(props->wiphy_freq), sizeof(uint32));
+	if (props->mac) {
+		netlink_attr(nlmsg, NL80211_ATTR_MAC, props->mac, ETH_ALEN);
+	}
+	if (props->wiphy_freq_fixed) {
+		netlink_attr(nlmsg, NL80211_ATTR_FREQ_FIXED, NULL, 0);
+	}
+	int err = netlink_send(nlmsg, sock);
+	if (err < 0) {
+		debug("join_ibss failed: %s\n", strerror(err));
+		return -1;
+	}
+	return 0;
+}
+
+static int get_ifla_operstate(struct nlmsg* nlmsg, const char* interface_name)
+{
+	struct ifinfomsg info;
+	memset(&info, 0, sizeof(info));
+	info.ifi_family = AF_UNSPEC;
+	info.ifi_index = if_nametoindex(interface_name);
+
+	int sock = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+	if (sock == -1) {
+		debug("get_ifla_operstate: socket failed: %d\n", errno);
+		return -1;
+	}
+
+	netlink_init(nlmsg, RTM_GETLINK, 0, &info, sizeof(info));
+	int n = 0;
+	int err = netlink_send_ext(nlmsg, sock, RTM_NEWLINK, &n);
+	close(sock);
+	if (err) {
+		debug("get_ifla_operstate: failed to query: %s\n", strerror(err));
+		return -1;
+	}
+
+	struct rtattr* attr = IFLA_RTA(NLMSG_DATA(nlmsg->buf));
+	for (; RTA_OK(attr, n); attr = RTA_NEXT(attr, n)) {
+		if (attr->rta_type == IFLA_OPERSTATE) {
+			return *((int32_t*)RTA_DATA(attr));
+		}
+	}
+
+	return -1;
+}
+
+static long syz_80211_join_ibss(volatile long a0, volatile long a1, volatile long a2, volatile long a3)
+{
+	uint8* ssid = (uint8*)a0;
+	int ssid_len = (int)a1;
+	int fix_frequency = (int)a2; // This parameter essentially determines whether it will perform a scan
+	int await_up = (int)a3;
+	struct nlmsg tmp_msg;
+	uint8 bssid[ETH_ALEN] = NETDEV_80211_IBSS_BSSID;
+
+	if (ssid == NULL || ssid_len < 0 || ssid_len > NETDEV_80211_MAX_SSID_LEN) {
+		// restricting SSID in order to avoid unnecessary segfaults on memcpy
+		return -1;
+	}
+
+	int sock = socket(AF_NETLINK, SOCK_RAW, NETLINK_GENERIC);
+	if (sock < 0) {
+		debug("syz_hwsim80211_join_ibss: socket failed: %d\n", errno);
+		return -1;
+	}
+
+	int nl80211_family_id = netlink_query_family_id(&tmp_msg, sock, "nl80211");
+	struct join_ibss_props ibss_props = {
+	    .wiphy_freq = NETDEV_80211_DEFAULT_FREQUENCY, .wiphy_freq_fixed = (fix_frequency > 0), .mac = bssid, .ssid = ssid, .ssid_len = ssid_len};
+
+	for (int device_id = 0; device_id < NETDEV_80211_INTITIAL_DEVICE_COUNT; device_id++) {
+		char interface[6] = {0};
+		memcpy(interface, "wlan0", 5);
+		interface[4] += device_id;
+
+		set_interface_state(interface, 0);
+		nl80211_set_interface(&tmp_msg, sock, nl80211_family_id, interface, NL80211_IFTYPE_ADHOC);
+		set_interface_state(interface, 1);
+		nl80211_join_ibss(&tmp_msg, sock, nl80211_family_id, interface, &ibss_props);
+
+		if (await_up) {
+			while (get_ifla_operstate(&tmp_msg, interface) != IF_OPER_UP)
```

Comment:
It waits ~ 9 seconds if wiphy_freq_fixed is false, and less than 0.5 seconds otherwise.

Definitions in `sys/linux/net_80211.txt` already take account of that.